Http/Controllers/API/V1/PriceController.php

```diff
@@ -23,75 +24,77 @@ public function index(Request $request)
             'query' => 'nullable|string|max:50',
         ]);
 
-        $lists = [];
-        if ($request->has('list_id')) {
-            $lists[] = $request->get('list_id');
-        } else if ($request->has('list_ids')) {
-            $lists = $request->array('list_ids');
-        } else if ($request->has('reimbursement_type_id')) {
-            $query = TenantPriceList::forReimbursementType($request->get('reimbursement_type_id'))
-                ->used();
+        $lists = $this->buildListsArrayFromRequestParams($request);
+        $authIssueResponse = $this->authorizationCheckForLists($lists);
+        if ($authIssueResponse) {
+            return $authIssueResponse;
+        }
 
-            if ($priceList = $query->first()) {
-                $lists[] = $priceList;
-            }
+        $query  = PriceQueryHelper::buildPriceOnlyQuery($request,$lists);
```

Comment:
The list building logic is now encapsulate in a method instead. I expect the changes to that code will have most impact on performance. More explanatory comments in the buildPriceOnlyQuery implementation further down.